### PR TITLE
AELO: let the user provide input data and receive feedback on field validation and job completion

### PR DESCRIPTION
**Pull request by ptormene**
The GUI is still very rough, but it already provides the main functionalities.
The concept of `APPLICATION_MODE` is introduced in `settings.py`, that can be set as `PUBLIC` (the basic webui without any authentication), `RESTRICTED` (setting `LOCKDOWN`=True and therefore requiring authentication), and `AELO` (requiring authentication and providing new AELO functionalities). The latter would add a very basic form to the webui, that for testing purposes is now pre-filled with some valid values for CCA. By pressing the `Submit` button, an ajax call is made to run the AELO calculation for the specified lon, lat, vs30 and siteid. In case the validation of the input values fails, a modal popup explains the validation error (it should be replaced by a better looking dialog with a title, or with some inline field validation messages). If the validation passes, the calculation is started. On failure or completion, an email is sent to the user that created the job (or the email is simulated, if the smtp backend is not configured). The email indicates the calculation id, the input values and, in case of failure, a description of the error, otherwise a link to retrieve the outputs.

The EngineServerTestCase was modified, inheriting from `django.test` instead of `unittest`, creating a test user in the `setUpClass` and doing the login with it, so it is possible to simulate the creation of the emails in the callbacks of AELO calculations.
A number of new tests were implemented, testing a successful AELO run, another AELO calculation that starts but fails afterwards because the given coordinates do not belong to any mosaic model, plus some cases in which the calculation does not start because of input validation failures.

I've factorized the successful aelo run procedure, so we can easily add tests for different mosaic models, just by adding one line with the corresponding input values (currently I've created one for CCA and one for EUR).

I've also made some changes to make sure that all test calculations are removed after they have been checked, otherwise it was annoying to run tests locally and have the local db filled by hundreds of test calculations after a few runs. Then I removed the final check that was making sure that at least one test calculation was completed (because all test calculations are supposed to be deleted at the time of that check).

### CODE REVIEW

File: openquake/engine/aelo.py [hunk 1]
```diff
@@ -73,7 +73,16 @@ def main(lon: valid.longitude,
...
             sys.exit('mosaic_dir is not specified in openquake.cfg')
-        jobctx.params.update(get_params_from(lon, lat, siteid))
+        try:
+            jobctx.params.update(get_params_from(lon, lat, siteid))
+        except ValueError as exc:
+            # FIXME: this can happen if no model covers the given coordinates.
+            #        In this case we need the job to fail and the email to be
+            #        sent. This is the only way I was able to make it happen,
+            #        but I guess there might be a better way to achieve the
+            #        same behavior
```
Comment (micheles) [inline openquake/engine/aelo.py:83]: This looks like the right way to do what you want

File: openquake/server/local_settings.py.template [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# This feature requires:
+#  - https://github.com/FirefighterBlu3/python-pam
+#  - https://github.com/cnobile2012/django-pam
+
```
Comment (micheles) [inline openquake/server/local_settings.py.template:4]: Why PAM? I absolutely don't want to have linux users on the server machine except `openquake`, we need to have pure Django authentication not associated to linux users.

File: openquake/server/tests/views_test.py [hunk 3]
```diff
@@ -122,8 +129,16 @@ def setUpClass(cls):
         env['OQ_DISTRIBUTE'] = 'no'
         # let's impersonate the user openquake, the one running the WebUI:
         # we need to set LOGNAME on Linux and USERNAME on Windows
-        env['LOGNAME'] = env['USERNAME'] = 'openquake'
+        env['LOGNAME'] = env['USERNAME'] = username = 'openquake'
```
Comment (micheles) [inline openquake/server/tests/views_test.py:132]: Let's create a Django user called "django-test-user" and not "openquake".

File: openquake/server/local_settings.py.template (hunk 2 above)
Comment (ptormene) [reply openquake/server/local_settings.py.template:4]: Ok, I'll create a template specifically for AELO, without PAM.

File: openquake/server/tests/views_test.py (hunk 3 above)
Comment (ptormene) [reply openquake/server/tests/views_test.py:132]: Ok

File: openquake/server/tests/views_test.py [hunk 4]
```diff
@@ -122,8 +129,16 @@ def setUpClass(cls):
...
+        cls.user, created = User.objects.get_or_create(
+            username=username, email=email)
+        if created:
+            cls.user.set_password(password)
+            cls.user.save()
         cls.c = Client()
+        cls.c.login(username=username, password=password)
 
     @classmethod
     def tearDownClass(cls):
```
Comment (ptormene) [inline openquake/server/tests/views_test.py:144]: We must delete the test user in tearDownClass

Comment (ptormene) [reply openquake/server/tests/views_test.py:144]: Done

File: openquake/server/tests/views_test.py (hunk 3 above)
Comment (ptormene) [reply openquake/server/tests/views_test.py]: Done

File: openquake/server/local_settings.py.template (hunk 2 above)
Comment (ptormene) [reply openquake/server/local_settings.py.template]: Done

**Review by micheles — APPROVED** [review]

**Review by micheles — APPROVED** [review]

**Pull request merged**